src/sync/client.rs

```diff
@@ -0,0 +1,109 @@
+use bson::Document;
+
+use super::Database;
+use crate::{
+    concern::{ReadConcern, WriteConcern},
+    error::Result,
+    options::{ClientOptions, DatabaseOptions, SelectionCriteria},
+    Client as AsyncClient,
+    RUNTIME,
+};
+
+/// This is the main entry point for the API. A `Client` is used to connect to a MongoDB cluster.
```

Comment:
Do we want to document anything about how this wraps the async client? If so, how detailed should we get about how we start the runtime under the hood?

Reply:
Yeah that sounds better, updated. Added the link for mutex as well as other internal links that were missing (e.g. to `mongodb::Client`).

This also made me realize I completely omitted the docstring for `sync::Database`, so I've added that as well.

Reply:
Looks good!